### PR TITLE
Add a gRPC logging extensions.

### PR DESCRIPTION
**Pull request by bianpengyuan**
To demonstrate gRPC and telemetry use case.

Signed-off-by: Pengyuan Bian <bianpengyuan@google.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,12 +5,12 @@ This repository contains several canonical Wasm extensions, which intend to demo
 * Development pattern of Wasm extension.
 * Best practice to test, build, and release a Wasm extension.
 
-# Extensions
-
-* [C++ empty scaffold](/extensions/scaffold/)
-* [Zig scaffold](/extensions/zig_demo/)
+# Extensio
```
Comment (kyessenov) [inline README.md:8]: typo: Extensions

File: extensions/grpc_logging/plugin.cc [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  getValue({"request", "url_path"}, &path);
+  getValue({"response", "code"}, &response_code);
+  getValue({"request", "referer"}, &referer);
+  getValue({"request", "user_agent"}, &user_agent);
+
+  *new_entry->mutable_timestamp() =
+      google::protobuf::util::TimeUtil::NanosecondsToTimestamp(timestamp);
+  *new_entry->mutable_latency() =
+      google::protobuf::util::TimeUtil::NanosecondsToDuration(duration);
+  new_entry->set_request_id(request_id);
```
Comment (kyessenov) [inline extensions/grpc_logging/plugin.cc:140]: You could just have passed `mutable_request_id` directly to `getValue`.

File: extensions/grpc_logging/plugin.cc [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+bool PluginRootContext::onDone() {
+  if (req_buffer_.empty() && in_flight_export_call_ == 0) {
+    return true;
+  }
+  // Flush out all log entries
+  flushLogBuffer();
+  sendLogRequest(/* ondone */ true);
+  return false;
```
Comment (kyessenov) [inline extensions/grpc_logging/plugin.cc:92]: I think it's worth adding a comment explaining that `proxy_done` will signal completion when `false` is returned.

**Review by kyessenov — APPROVED** [review]
Minor comments.

File: extensions/grpc_logging/plugin.cc (hunk 2 above)
Comment (bianpengyuan) [reply extensions/grpc_logging/plugin.cc]: Much better! Thanks!

**Pull request merged**